packages/ui/routes/mailings/__partials/mailings-selection-actions.vue

```diff
@@ -96,6 +143,56 @@ export default {
       }
       this.closeMoveManyMailsDialog();
     },
+    handleInitDownload({
+      withFtp,
+      mailingIds,
+      havePreview,
+      mailsWithoutPreviewSelection,
+    }) {
+      if (havePreview) {
+        this.handleDownloadMailSelections({ withFtp, mailingIds });
+        return;
+      }
+      this.$refs.downloadMailsDialog.open({
+        withFtp,
+        mailingIds,
+        mailsWithoutPreviewSelection,
+      });
+    },
+    handleInitSingleDownload({ withFtp, mailing }) {
+      const mailingIds = [mailing.id];
+      const havePreview = mailing.hasHtmlPreview;
```

Comment:
intead of adding new const we can just edit the last one. **you dont have to change it, as u like** 
```jsx
const {hasHtmlPreview: hasPreview} = mailing
```

Reply:
`const {hasHtmlPreview: hasPreview} = mailing` does not edit but add a new const same as i did that why you write **const**  before declaration, so it s the same thing as `const havePreview = mailing.hasHtmlPreview;` your feedback would have been good if i had extracted more varaibles but in my case i just used one